### PR TITLE
FDC3 2.0 Adds/Edits/Deletes

### PR DESCRIPTION
**Pull request by bingenito**

### CODE REVIEW

File: src/Fdc3/Context/Currency.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+            : base(ContextTypes.Currency, id, name)
+        {
+        }
+
+        object? IContext<object>.ID => base.ID;
+    }
+
+    public class CurrencyID
+    {
+        public string? CURRENCY_ISOCODE { get; set; }
```
Comment (psmulovics) [inline src/Fdc3/Context/Currency.cs:29]: Was this fixed in the end?

File: src/Fdc3/IIcon.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+{
+    public interface IIcon
+    {
+        /// <summary>
+        /// The icon url
+        /// </summary>
+        string Src { get; }
+        
+        /// <summary>
+        /// The icon dimensions</height>'
```
Comment (psmulovics) [inline src/Fdc3/IIcon.cs:25]: </height> ?

File: src/Fdc3/IImage.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+{
+    public interface IImage
+    {
+        /// <summary>
+        /// The icon url
+        /// </summary>
+        string Src { get; }
+        
+        /// <summary>
+        /// The icon dimensions</height>'
```
Comment (psmulovics) [inline src/Fdc3/IImage.cs:25]: `</height>` ?

File: src/Fdc3/IIntentResult.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+ */
+
+using MorganStanley.Fdc3.Context;
+
+namespace MorganStanley.Fdc3
+{
+    /// <summary>
+    /// Intents can return results that are either context data objects or a reference to a Channel.
+    /// </summary>
+    public interface IIntentResult {
```
Comment (psmulovics) [inline src/Fdc3/IIntentResult.cs:22]: Do we need to add a `#pragma` here?

**Review by psmulovics — CHANGES_REQUESTED** [review]
Left a few code smelly comments

File: src/Fdc3/Context/Currency.cs (hunk 1 above)
Comment (bingenito) [reply src/Fdc3/Context/Currency.cs:29]: That was on valuation, and yes - https://github.com/finos/FDC3/issues/833

File: src/Fdc3/IIcon.cs (hunk 2 above)
Comment (bingenito) [reply src/Fdc3/IIcon.cs]: Copy paste and autocorrect thinking it was html or xml in the comments really butchered it.

**Review by psmulovics — APPROVED** [review]

**Comment by bingenito** [comment]
Fixes #4

**Pull request merged**